### PR TITLE
⚠️ Move `.spec.install.namespace` and `.spec.install.serviceAccount` to `.spec.namespace` and `.spec.serviceAccount`

### PR DESCRIPTION
**Pull request by everettraven**
# Description
Move `.spec.install.namespace` and `.spec.install.serviceAccount` to `.spec.namespace` and `.spec.serviceAccount`, updates descriptions of those fields, and makes `.spec.install` optional. 

Immediate follow up to #1404 

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: api/v1alpha1/clusterextension_types.go [hunk 1]
```diff
@@ -43,6 +43,34 @@ const (
 
 // ClusterExtensionSpec defines the desired state of ClusterExtension
 type ClusterExtensionSpec struct {
+	// namespace is a reference to a Kubernetes namespace.
+	// This is the namespace the provided ServiceAccount must exist.
```
Comment (joelanford) [inline api/v1alpha1/clusterextension_types.go:47]:
```suggestion
	// This is the namespace in which the provided ServiceAccount must exist.
```

File: internal/controllers/clusterextension_admission_test.go [hunk 2]
```diff
@@ -402,6 +388,66 @@ func TestClusterExtensionAdmissionServiceAccount(t *testing.T) {
...
+func TestClusterExtensionAdmissionInstall(t *testing.T) {
+	oneOfErrMsg := "at least one of [preflight] are required when install is specified"
+
+	testCases := []struct {
+		name          string
+		installConfig *ocv1alpha1.ClusterExtensionInstallConfig
+		errMsg        string
+	}{
+		{
+			"install specified, nothing configured",
```
Comment (joelanford) [inline internal/controllers/clusterextension_admission_test.go:400]: Add field names to the struct instances?

Comment (everettraven) [reply internal/controllers/clusterextension_admission_test.go]: https://github.com/operator-framework/operator-controller/pull/1439/commits/1685475e0d8d5aac66e7abf7388f31b09bc7525c

File: api/v1alpha1/clusterextension_types.go (hunk 1 above)
Comment (everettraven) [reply api/v1alpha1/clusterextension_types.go]: https://github.com/operator-framework/operator-controller/pull/1439/commits/1685475e0d8d5aac66e7abf7388f31b09bc7525c

**Review by LalatenduMohanty — DISMISSED** [review]
/lgtm

**Comment by joelanford** [comment]
Any docs updates necessary for this change?

**Comment by everettraven** [comment]
> Any docs updates necessary for this change?

Ah, probably. I'll look into this on Monday.

**Review by m1kola — APPROVED** [review]
Looks good to me.

I see there is a question about docs - I had a look and our docs are broken already unfortunately (even [getting started](https://operator-framework.github.io/operator-controller/getting-started/olmv1_getting_started/) page). It is because previous changes were not reflected in the docs (e.g. `pollInterval` being renamed to `pollIntervalMinutes`).

I'm going to merge this PR as is and will go through the docs in the afternoon.

**Comment by grokspawn** [comment]
Related to #1426

**Pull request merged**